modules/assim.sequential/R/Adjustment.R

```diff
@@ -0,0 +1,57 @@
+##' @title adj.ens
+##' @name  adj.ens
+##' @author Michael Dietze \email{dietze@@bu.edu}, Ann Raiho and Hamze Dokoohaki
+##' 
+##' @param Pf  A cov matrix of forecast state variables.  
+##' @param X   Dataframe or matrix of forecast state variables for different ensembles.
+##' @param X.new Dataframe or matrix of forecast state variables for different ensembles. When processvar is TRUE.
+##' @param mu.f A vector with forecast mean estimates of state variables.
+##' @param mu.a A vector with analysis mean estimates of state variables.
+##' @param Pa The state estimate cov matrix of analysis.
+##' @param processvar Boolean flag for indicating if process variance is used in the analysis. If TRUE then X.new needs to be supplied.
+##’ @details
+##’  
+##' 
+##' @description This functions gives weights to different ensemble members based on their likelihood during the analysis step. Then it adjusts the analysis mean estimates of state variables based on the estimated weights.
+##' 
+##' @return Returns a vector of adjusted analysis mean estimates of state variables.
+##' @export
+
+adj.ens<-function(Pf,X,X.new,mu.f,mu.a,Pa,processvar){
+
+  S_f  <- svd(Pf)
+  L_f  <- S_f$d
+  V_f  <- S_f$v
+  
+  ## normalize
+  Z <- X*0
+  
+  for(i in seq_len(nrow(X))){
+    if(processvar == TRUE) {
+      Z[i,] <- 1/sqrt(L_f) * t(V_f)%*%(X.new[i,]-mu.f)
+    }else{
+      Z[i,] <- 1/sqrt(L_f) * t(V_f)%*%(X[i,]-mu.f)
+    }
+  }
+  Z[is.na(Z)]<-0
+  
+  ## analysis
+  S_a  <- svd(Pa)
+  
+  L_a  <- S_a$d
+  V_a  <- S_a$v
+  
+  ## analysis ensemble 
+  X_a <- X*0
+  for(i in seq_len(nrow(X))){
+    # she decomposed Pa - then it's putting it back together but with a different Z which comes from the likelihood of that ens    X_a[i,] <- V_a %*%diag(sqrt(L_a))%*%Z[i,] + mu.a
```

Comment:
is there a reason why this is turned off? or if it's not needed anymore is there a reason why the loop left open?